code-dot-org/code-dot-org
# [CSC/Artist] Add "draw shape" blocks

Amy is looking to add new "stickers" to Artist for a CSC lesson. These new stickers need to be added ASAP for levels to be developed in time for an upcoming pilot. This issue is labeled as a high priority.
CSC requires a new set of rhombus images (from https://github.com/code-dot-org/code-dot-org/pull/45408) that do not include any of the current Artist stickers. We also don't want to flood the existing `draw sticker` blocks with a bunch of rhombuses.

The solution explored here proposes the creation of 3 new blocks for Artist (called `draw shape` that are based on the 3 existing `draw sticker` blocks. The key differences are:
* A completely separate list of new images.
* Stickers are scaled by absolute image size. Shapes are scaled by the length of the sides of the rhombuses that are pictured. Any other shape images added to this block (triangles, hexagons, etc.) should follow this same design convention. The canvas was sized around the shapes to allow for this eventuality.
* A higher maximum size.

These blocks will not be added to CSF or the Artist project level at this time. This PR also includes a new artist level pre-populated with the new blocks to help the CSC team begin levelbuilding as soon as possible.

*New Blocks:*
<img width="442" alt="image" src="https://user-images.githubusercontent.com/43474485/159764831-67010461-a26c-4a91-9fd5-15d65e9dcefc.png">

*The following screenshots were taken when the blocks were called `draw pattern block`. It has since been renamed to `draw shape` to avoid overloading the word "block." (Change approved by AmyBW)*

*Shapes as they might show up in the curriculum:*
![image](https://user-images.githubusercontent.com/43474485/159515895-fe59aafe-18c2-4031-b5d3-a5cfcaa0f8f3.png)

*Tracing Shapes to show how `side length` relates to other artist movements:*
![image](https://user-images.githubusercontent.com/43474485/159517493-5fddbb13-b43a-414f-90a8-cbc0f201df52.png)

*Comparing Shapes with Stickers:*
Stickers default to a `size` of 100 (max width or height). Shapes default to `side length` of 100, which is why the images appear larger.
![image](https://user-images.githubusercontent.com/43474485/159516717-4abb6c1e-295e-4ac1-9801-3bbe53aeafc9.png)


## Links

- jira ticket: [STAR-2163: [CSC/Artist] Sticker Addition](https://codedotorg.atlassian.net/browse/STAR-2163)

## Testing

**To test these changes:**
* Pull changes
* Reseed levels locally
* Find the new level while running your dashboard server: http://localhost-studio.code.org:3000/levels?utf8=%E2%9C%93&name=pattern_blocks_sandbox
* You should see the new images and block in an Artist level

## Follow-up work

Amy and Kylie to collaborate with Mark/Whitney to create more polished images that can replace the ones here after the pilot. The number of available images might also change would require updating the list found in `apps/src/turtle/skins.js`.
